src/update.c

```diff
@@ -365,7 +365,45 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Step 4: Download packs */
+
+	/* Delta manifests may be included in the delta packs, so the packs must be downloaded
+	 * before loading subscribed manifests. */
+	latest_subs = list_clone(current_subs);
+	set_subscription_versions(server_manifest, current_manifest, &latest_subs);
+	link_submanifests(current_manifest, server_manifest, current_subs, latest_subs, false);
+
+	/* The new subscription is seeded from the list of currently installed bundles
+	 * This calls add_subscriptions which recurses for new includes */
+	grabtime_start(&times, "Add Included Manifests");
+	ret = add_included_manifests(server_manifest, current_version, &latest_subs);
+	grabtime_stop(&times);
+	if (ret) {
+		ret = EMANIFEST_LOAD;
+		goto clean_exit;
+	}
+	set_subscription_versions(server_manifest, current_manifest, &latest_subs);
+
+	grabtime_start(&times, "Download Packs");
```

Comment:
The "Recurse and Consolidate Manifests" time recording block starts before loading the server MoM, so now it includes pack downloading in its time statistics. I think that the "Recurse and Consolidate Manifests" time block should be moved so that it starts before loading the subscribed manifests instead of before loading the server MoM. I will open a separate issue for this.

Reply:
Below is the timing issue:

#584